laws/src/main/scala/cats/laws/discipline/Arbitrary.scala

```diff
@@ -276,6 +276,20 @@ object arbitrary extends ArbitraryInstances0 {
   implicit def catsLawsCogenForAndThen[A, B](implicit F: Cogen[A => B]): Cogen[AndThen[A, B]] =
     Cogen((seed, x) => F.perturb(seed, x))
 
+  implicit def catsLawsArbitraryForChain[A](implicit A: Arbitrary[A]): Arbitrary[Chain[A]] =
+    Arbitrary(Gen.sized {
+      case 0 => Gen.const(Chain.nil)
+      case 1 => A.arbitrary.map(Chain.one)
+      case 2 => A.arbitrary.flatMap(a1 => A.arbitrary.flatMap(a2 =>
+        Chain.concat(Chain.one(a1), Chain.one(a2))))
+      case n => Chain.fromSeq(Range.apply(0, n)).foldLeft(Gen.const(Chain.empty[A])) { (gen, _) =>
```

Comment:
I don’t know what the fromSeq is doing here because we are just folding it. I don’t love this generator because it does not generate any internal Wrap nodes and is always left associated.

Could we do something like: for size n we either wrap a seq or create a two random chains of size a and b such that a + b == n and we combine those.

If we do this, I think we can get the coverage up nice and high.